### PR TITLE
release 0.6.4

### PR DESCRIPTION
**Pull request by arnoFleming**
Add a changelog, link to it from the gemspec

### CODE REVIEW

File: wetransfer_style.gemspec [hunk 1]
```diff
@@ -1,15 +1,22 @@
...
+  s.version = "0.6.4"
   s.summary = "At WeTransfer we code in style. This is our style."
   s.description = s.summary
   s.homepage = "https://github.com/WeTransfer/wetransfer_style"
   s.license = "MIT"
 
   s.files = Dir["README.md", "LICENSE", "ruby/*.yml"]
 
-  s.add_dependency "rubocop", "0.52.1"
+  # lets not allow a major bump for Rubocop, but there is nothing special about 0.61.1
```
Comment (linkyndy) [inline wetransfer_style.gemspec]: This is major and minor, just to make sure.

Not sure how rubocop does updates, but we might stick to a fixed version, since any _slight_ change can break the projects' style that use this gem.

Comment (arnoFleming) [reply wetransfer_style.gemspec]: Thanks for your input. 

I am glad to see the fixed version discussion! I'm just not sure if it belongs in this repo. 

I think it would make sense to pin to a specific version in the repo that uses this gem. 

That means this gem can be optimistic about rubocop, so some projects can use (and pin) `0.61.1`, while other projects can use `0.62.xyz` whenever it is released, without this gem needing an update. 

Could that work?

Comment (arnoFleming) [reply wetransfer_style.gemspec]: You write 

> This is major and minor, just to make sure.

Please give more context, I am unsure if I understand it.

Comment (linkyndy) [reply wetransfer_style.gemspec]: Hmm, we started with the idea of only including this gem in other projects, and not rubocop manually. That way we can be _sure_ of the style enforced by a specific version of this gem (which was the end goal, eliminating any ambiguity). This is still my take on the subject.

Comment (linkyndy) [reply wetransfer_style.gemspec]: Sorry. `~> 0.61.1` resolves to `0.61.x`, where `x >= 1`. So, only the patch version is changed (major and minor remain the same).

Comment (arnoFleming) [reply wetransfer_style.gemspec]: Thanks for the clarification. The `~> 0.61.1` was intended as `~> 0.61`. Good catch. 

But reading their [installation notes](https://github.com/rubocop-hq/rubocop#installation) seems to resolve this discussion: 

> RuboCop's development is moving at a very rapid pace and there are often backward-incompatible changes between minor releases (since we haven't reached version 1.0 yet). To prevent an unwanted RuboCop update you might want to use a conservative version locking in your `Gemfile`:
> 
> `gem 'rubocop', '~> 0.62.0', require: false`

Would leaving the version specification like this resolve your tensions?

Comment (linkyndy) [reply wetransfer_style.gemspec]: I considered it was better to be safe than sorry, that's why I went for the hard version; you never know how gem authors bump versions. I also had a discussion with @julik and we settled on this. But I'm not against this change; we can try it and if things don't break unexpectedly, it's fine 👍

**Comment by arnoFleming** [comment]
@linkyndy, @julik please 👍 or 👎 this whole thing. Maybe suggest changes in the latter case

**Comment by julik** [comment]
TL;DR - Yes, I think wetransfer_style _must_ lock rubocop exactly.

I think you might have a problem with this approach. See, rubocop (being an Eastern-European product made in a post-totalitarian country) switches the rules on you every so often. Every so often can be any minor version. The rules are also Ruby-version dependent sometimes. I don't have specific links at the moment but you will find aplenty on the rubocop bugtracker.

What this means in practice is that the output of rubocop with one and the same wetransfer_style version can be different. It will depend on whichever version of rubocop that gets installed according to the gem specification - and since quite a few things using wetransfer_style are libraries, they do not have a Gemfile.lock. Meaning they will grab _any_ rubocop that vaguely matches the version definition. Meaning that consequently _all_ the contributions to those libraries depend on whichever version of rubocop gets downloaded during the specific build on CI you can get flagged for bad formatting or not.

Which is exactly what we should avoid at all costs (having "rules change underneath you"), since (as befits people from post-totalitarian countries) I can tell you it doesn't feel nice at all.

Basically...

> there is nothing special about 0.61.1

Except did you test? With all the libraries that use wetransfer_style, with all the Ruby versions they have to be compatible with?

File: wetransfer_style.gemspec (hunk 1 above)
Comment (julik) [reply wetransfer_style.gemspec]: > I think it would make sense to pin to a specific version in the repo that uses this gem.

Nope. See below.

**Comment by linkyndy** [comment]
@julik gave a thorough explanation of my thoughts as well 👍

**Review by julik — CHANGES_REQUESTED** [review]
We need to pin rubocop.

**Comment by arnoFleming** [comment]
I think it is fair to only use tested dependecies, but disapproving updates from post-totalitarian countries while using prior versions of their software feels like a logical fallacy. I'm not saying your judgement is wrong, I just don't see the logic behind it.

Also; I understand that we use wetransfer_style from different ruby (and rails) apps and libraries. . Both `Gemfile`s and `gemspec`s can pin versions: https://guides.rubygems.org/specification-reference/#add_runtime_dependency 

For me that feels like a fantastic way to let the cool kids (emerging products / innovation friday things) use newe(r|st) versions from _not-invented-here_ libraries, while the battle tested repo's can pin a specific version of both wetransfer_style and rubocop.

Otherwise, a fork of this repo would be needed just to use an update of a dependency of this library,  that feels subobtimal to me.

**Comment by linkyndy** [comment]
I see your point. But the idea here is to establish a style that should be used _worldwide_ (or rather, repowide). That can only be done by providing this gem with a **specific** style, i.e.: a specific version of Rubocop. Otherwise, we're not offering a style, but a mixed pot of styles, which is what was happening at the moment we decided to create this gem.

Specific repos can pin Rubocop separately, but then how is this gem offering a uniform style across our repos?

It's a matter of style, so being always up-to-date with the gem isn't a thing we strive for; cool kids can (and should) use the same style, if they want to adhere to WeTransfer style.

**Comment by julik** [comment]
The logic is terse here - ssumes the following two requirements:

* Do not make your user feel stupid.
* Do not make your user do shit work.

The user in that case is a maintainer of a codebase relying on wetransfer_style for linting. Specializing from the above principles, we can further define what we want from a linter:

* If we want to have a "baseline" on code style, checking this code style should be easy.
* Checking the code style should be extremely consistent between lint runs, it may not depend on an arbitrary change of an arbitrary peer dependency
* Installing the linter and ensuring it works correctly should be a non-issue. If you ask the person opting into wetransfer_style to pin rubocop themselves you are saying "Oh, here is a linter you kinda have to conform to, but if you want to do that - sucks to be you, now go figure out why the linter doesn't work with the particular combination of the Ruby version and rubocop that you are using." 
* Corollary to the above: there should be no configuration, or the absolute possible minimum of configuration. If you ask me the fact that we need a rubocop.yml in the repo root for the linter to work it is already too much work 🤔

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,8 +2,18 @@
 
 ## 0.6.4
 
-Add changelog, homepage and source code metadata so Rubygems can create links
+[Added]
+
+- This changelog file
+- Rubygems metadata in gemspec so other libs can parse this changelog
```
Comment (linkyndy) [inline CHANGELOG.md:8]: Not sure how useful these entries are. I would keep the changelog as short as possible, containing only the _really_ important things.

**Review by julik — APPROVED** [review]

File: CHANGELOG.md (hunk 2 above)
Comment (arnoFleming) [reply CHANGELOG.md:8]: Well, it is important, since it solves an issue: 

The objection was (see below) that a changelog could not be found. We now not only have one, but it is accessible in a standardized way. 

> screenshot

**Comment by julik** [comment]
🍾
